test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -989,6 +989,8 @@ def _revert_pbsconf_comm(self, primary_server, vals_to_set):
                 self.du.set_pbs_config(comm.hostname, confs=new_pbsconf)
                 comm.pbs_conf = new_pbsconf
                 comm.pi.initd(comm.hostname, "restart", daemon="comm")
+                if not comm.isUp():
+                    self.fail("comm is not up")
```

Comment:
instead of fail() Can you update it as follows

                rv = comm.isUp()
                if not rv:
                    self.logger.error('comm ' + comm.hostname + ' is down')
                    comm.start()
                    msg = 'Failed to restart comm ' + comm.hostname
                    self.assertTrue(comm.isUp(), msg)

Applies to other daemons.

Reply:
but we already do a comm restart/start just before calling comm.isUp(), will it really help to try it again?

Reply:
Nothing specific just wanted to retry before exiting.

Reply:
I'm not sure if running the same command again will help, but I think it might help if we modify isUp to wait and poll for a few times before giving up. Mom's isUp already does this:

        for _ in range(max_attempts):
            rv = super(MoM, self)._isUp(self)
            if rv:
                break
            time.sleep(1)

If this sounds like a good idea for all isUp functions, then maybe we should add this to __isUp() itself